services/src/main/java/io/scalecube/services/ServiceCall.java

```diff
@@ -36,56 +37,69 @@
   private final ServiceRegistry serviceRegistry;
   private final Router router;
 
+  private final Function<ServiceMessage, Object> msgToResp =
+      sm -> sm.hasData() ? sm.data() : EMPTY_RESPONSE_ERROR;
+
+  private final Function<Boolean, Function<Mono<ServiceMessage>, Mono<Object>>> monoRespFn =
+      isServiceMsg ->
+          isServiceMsg
+              ? mono -> mono.cast(Object.class)
+              : mono -> mono.map(msgToResp);
+
+  private final Function<Boolean, Function<Flux<ServiceMessage>, Flux<Object>>> fluxRespFn =
+      isServiceMsg -> isServiceMsg ? flux -> flux.cast(Object.class) : flux -> flux.map(msgToResp);
+
   private ServiceCall(Call call) {
     this.transport = call.transport;
     this.methodRegistry = call.methodRegistry;
     this.serviceRegistry = call.serviceRegistry;
     this.router = call.router;
   }
 
-  /**
-   * This class represents {@link ServiceCall}'s definition. All {@link ServiceCall} must be created
-   * out of this definition.
-   */
-  public static class Call {
-
-    private Router router = Routers.getRouter(RoundRobinServiceRouter.class);
-
-    private final ClientTransport transport;
-    private final ServiceMethodRegistry methodRegistry;
-    private final ServiceRegistry serviceRegistry;
-
-    /**
-     * Creates new {@link ServiceCall}'s definition.
-     *
-     * @param transport - transport to be used by {@link ServiceCall} that is created form this
-     *     {@link Call}
-     * @param methodRegistry - methodRegistry to be used by {@link ServiceCall} that is created form
-     *     this {@link Call}
-     * @param serviceRegistry - serviceRegistry to be used by {@link ServiceCall} that is created
-     *     form this {@link Call}
-     */
-    public Call(
-        ClientTransport transport,
-        ServiceMethodRegistry methodRegistry,
-        ServiceRegistry serviceRegistry) {
-      this.transport = transport;
-      this.serviceRegistry = serviceRegistry;
-      this.methodRegistry = methodRegistry;
+  private static ServiceMessage toServiceMessage(MethodInfo methodInfo, Object... params) {
```

Comment:
Please remove `static` modifier and move at the end of the class, and the same for `toServiceMessage`, `noReachableMemberException`, `toStringOrEqualsOrHashCode`. See `addressLookup`.